# Pandas Time Series Exercise Set #1

For this set of exercises we'll use a dataset containing monthly milk production values in pounds per cow from January 1962 to December 1975.

<div class="alert alert-danger" style="margin: 10px"><strong>IMPORTANT NOTE!</strong> Make sure you don't run the cells directly above the example output shown, <br>otherwise you will end up writing over the example output!</div>

In [1]:
# RUN THIS CELL
import pandas as pd
%matplotlib inline

df = pd.read_csv('../Data/monthly_milk_production.csv', encoding='utf8')
title = "Monthly milk production: pounds per cow. Jan '62 - Dec '75"

print(len(df))
print(df.head())

168
      Date  Production
0  1962-01         589
1  1962-02         561
2  1962-03         640
3  1962-04         656
4  1962-05         727


So <tt>df</tt> has 168 records and 2 columns.

### 1. What is the current data type of the Date column?
HINT: We show how to list column dtypes in the first set of DataFrame lectures.

In [2]:
# CODE HERE


In [3]:
# DON'T WRITE HERE

### 2. Change the Date column to a datetime format

In [4]:
# DON'T WRITE HERE

### 3. Set the Date column to be the new index

In [5]:
# DON'T WRITE HERE

### 4. Plot the DataFrame with a simple line plot. What do you notice about the plot?

In [6]:
# DON'T WRITE HERE

### 5. Add a column called 'Month' that takes the month value from the index
HINT: You have to call <tt>df.index</tt> as <tt>df['Date']</tt> won't work.

<strong>BONUS: See if you can obtain the <em>name</em> of the month instead of a number!</strong>

In [7]:
# DON'T WRITE HERE

In [8]:
# BONUS SOLUTION:

### 6. Create a BoxPlot that groups by the Month field

In [9]:
# DON'T WRITE HERE

# Great Job!